src/checker/zx/ZXChecker.cpp

```diff
@@ -129,6 +129,23 @@ qc::Permutation complete(const qc::Permutation& p, const std::size_t n) {
     mappedTo[v]   = true;
   }
 
+  // Map rest greedily
+
+  for (std::size_t i = 0; i < n; ++i) {
+    if (mappedFrom[i]) {
+      continue;
+    }
+
+    for (std::size_t j = 0; j < n; ++j) {
+      if (!mappedTo[j]) {
+        pComp[static_cast<qc::Qubit>(i)] = static_cast<qc::Qubit>(j);
+        mappedTo[j]                      = true;
+        mappedFrom[i]                    = true;
+        break;
+      }
+    }
+  }
+
```

Comment:
Since this has been moved. The comment does not really fit too well anymore.
Was there a particular reason for moving this block of code?

Reply:
That code was precisely the problem. This completion of an incomplete permutation just worked in the previous test cases. It turns out that the choice of permutation is completely arbitrary. Because of a bug in the choice the segfault occured.